modules/rostests/apitests/kernel32/Pipes.c

```diff
@@ -0,0 +1,131 @@
+/*
+ * PROJECT:         ReactOS api tests
+ * LICENSE:         GNU GPLv2 only as published by the Free Software Foundation
+ * PURPOSE:         Test for pipe (CORE-17376)
+ * PROGRAMMER:      Simone Mario Lombardo
+ */
+
+#include "precomp.h"
+
+#define LP TEXT("\\\\.\\pipe\\rostest_pipe")
+
+#define MINBUFFERSIZE 1
+#define MAXBUFFERSIZE 255
+
+static DWORD dReadBufferSize;
+
+DWORD
+WINAPI
+PipeWriter(
+        _In_ PVOID Param)
+{
+    DWORD cbWritten;
+    HANDLE hPipe;
+    DWORD dwLastError;
+    BOOL Success;
+    PCSTR pszInMsg = "Test";
+
+    hPipe = CreateFile(LP, GENERIC_READ | GENERIC_WRITE, FILE_SHARE_READ | FILE_SHARE_WRITE,
+            NULL, OPEN_EXISTING, FILE_ATTRIBUTE_NORMAL, NULL);
+    ok(hPipe != INVALID_HANDLE_VALUE, "CreateFile failed, results might not be accurate\n");
+    if (hPipe != INVALID_HANDLE_VALUE)
+    {
+        Sleep(0);
+        Success = WriteFile(hPipe, &pszInMsg, 4, &cbWritten, NULL);
+        dwLastError = GetLastError();
+
+        ok(Success, "Pipe's WriteFile returned FALSE, instead of expected TRUE\n");
+        if(dwLastError != 0)
+            trace("Last Error = 0x%lX\n",dwLastError);
+
+        ok(cbWritten > 0,"Pipe's Writefile has lpNumberOfBytesWritten <= 0\n");
+
+        CloseHandle(hPipe);
+    }
+    return 0;
+}
+
+DWORD
+WINAPI
+PipeReader(
+        _In_ PVOID Param)
+{
+    HANDLE hPipe;
+    DWORD cbRead;
+    HANDLE hThread;
+    DWORD dwLastError;
+    BOOL Success;
+    CHAR szOutMsg[MAXBUFFERSIZE];
+
+    hPipe = CreateNamedPipeA( 
+        LP,
+        PIPE_ACCESS_DUPLEX,
+        PIPE_TYPE_MESSAGE|PIPE_READMODE_MESSAGE,
+        1,
+        MAXBUFFERSIZE,
+        MAXBUFFERSIZE,
+        0,
+        NULL
+    );
+    ok(hPipe != INVALID_HANDLE_VALUE, "CreateNamedPipeA failed\n");
+    if (hPipe != INVALID_HANDLE_VALUE)
+    {
+        hThread = CreateThread(NULL,0, PipeWriter, NULL, 0, NULL);
```

Comment:
Same as with `CreateNamedPipeA`: Would be better readable (less convoluted) if this was done in `StartTestCORE17376`.

Reply:
Okkay, I will try ASAP.